### DISCUSSION TITLE
Is no returning double nil really a good idea?

### DISCUSSION THREAD

**Discussion by zhouzilong2020** [benbjohnson/wtf#71] [General]
There are some senarios that returning an error when FindXXXXX actually get noting is confusing:

1. Say you want to have a function with the semantic that first check whether XXXXX is exist, if not, create one.
2. If a function returns a pointer, it is always nice to check the whether it is null first before using it, thus leviating the potential panic issue.
3. The caller have to deal the existence issue by checking the error type, rather than checking the pointer returned.

**Comment by benbjohnson** [comment]
You can return double nil. It's mostly a matter of preference. Originally, that's what I did but I found that it was easy to forget to check for double nil whereas checking for errors is very common in Go.
